### ISSUE TITLE
Support for CosmosDbs in azurerm_security_center_subscription_pricing

### ISSUE THREAD

**Issue by dvrmsy** [hashicorp/terraform-provider-azurerm#18033]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

We use the azurerm_security_center_subscription_pricing to set enable Defender features however Cosmos is not supported currently. 

From examining the REST API documentation for **Microsoft.Security/pricings** it appears the value should be CosmosDbs however trying this value generates an error **Error: expected resource_type to be one of**  followed by a list matching the currently supported options in the terraform documentation

### New or Affected Resource(s)/Data Source(s)

azurerm_security_center_subscription_pricing

### Potential Terraform Configuration

```hcl
resource "azurerm_security_center_subscription_pricing" "example" {
  tier          = "Standard"
  resource_type = "CosmosDbs"
}
```

**Comment by websolut** [comment]
Any update on this issue - it is not a subplan I am afraid.

**Comment by simschere** [comment]
@websolut indeed, it is considered as a resource_type by Microsoft. Let's hope it will be supported soon by Terraform.

**Comment by sreallymatt** [comment]
Closed by #20761
